policyengine_api/endpoints/policy.py

```diff
@@ -198,45 +206,66 @@ def get_policy_search(country_id: str) -> list:
     if country_not_found:
         return country_not_found
 
-    results = database.query(
-        "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
-        (country_id, f"%{query}%"),
-    )
-    if results is None:
-        return dict(
-            status="error",
-            message=f"Policy not found in {country_id}",
+    try:
+        results = database.query(
+            "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
+            (country_id, f"%{query}%"),
         )
-    else:
+        if results is None:
+            body = dict(
+                status="error",
+                message=f"Database query failed for {country_id} with query '{query}",
```

Comment:
```suggestion
                message=f"Policy not found in {country_id}",
```
I would actually revert this message and the status code below. A 500 represents a server error - for example, if the server crashed because something was formatted incorrectly. In this case, a record merely doesn't exist, which typically signals a 404 being the correct type of error.

Reply:
Perhaps remove the `result is None` conditional altogether, since it does not catch empty lists, and just leave `not results` ? then  and catch any data retrieval issues in the `try` block

Reply:
Good point, that is one of the cases that would not be caught by `result is None`. Yeah, I think your idea is the best one.